### PR TITLE
Query: Include collection and order by subquery throws 'Variable names must be unique within a query batch or stored procedure.'

### PR DESCRIPTION
**Pull request by Tasteful**
Replaces PR #5405 (that I messad up when syncing code with dev)

@anpete @smitpatel

### CODE REVIEW

**Comment by Tasteful** [comment]
Moving comment from replaced PR: Fix for #4521 Potential fix for #5038 & #4695

**Comment by Tasteful** [comment]
@anpete @smitpatel The test that is failing is working when running on Windows 10 so don't know what it can be. Is it ppssible to trigger the AppVeyor to build this?

**Comment by smitpatel** [comment]
@Tasteful - That test failure is due to my recent changes. I am looking into it. It did work for me on windows but failing on Linux. (strange! 😨 )
Generally to trigger a new build on PR, you need to add commit the pull request. (even blank commit or amended commit can do)

**Comment by smitpatel** [comment]
@Tasteful - Linux issue got fixed. PR is passing successfully now. Can you add Assertions in `IncludeSqlServerTest` to verify the generated Sql.
Also can you squash all commits into one.

**Comment by Tasteful** [comment]
@smitpatel `IncludeSqlServerTest` added and also squashed all the commits into one.

**Comment by smitpatel** [comment]
@anpete - for review.

File: src/Microsoft.EntityFrameworkCore.Relational/Query/ExpressionVisitors/Internal/IncludeExpressionVisitor.cs [hunk 1]
```diff
@@ -378,7 +378,20 @@ var newJoinExpression
             SelectExpression targetSelectExpression,
             TableExpressionBase innerJoinExpression)
         {
-            foreach (var ordering in innerJoinSelectExpression.OrderBy)
+            var needOrderingChanges = innerJoinSelectExpression.OrderBy.Any(x => x.Expression is SelectExpression || x.Expression.IsAliasWithColumnExpression());
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/ExpressionVisitors/Internal/IncludeExpressionVisitor.cs]: nit: break this long line

**Comment by anpete** [comment]
:shipit:

**Comment by smitpatel** [comment]
merged via c97c523c179df16600528d67e5048899f0131c22

**Comment by smitpatel** [comment]
@Tasteful - Thank you for contribution.

**Pull request closed without merging**